### PR TITLE
Implement document highlight fallback for search-based intel

### PR DESCRIPTION
**Pull request by olafurpg**
Fixes https://github.com/sourcegraph/sourcegraph/issues/35843

### CODE REVIEW

File: template/src/search/providers.ts [hunk 1]
```diff
@@ -292,7 +292,38 @@ export function createProviders(
...
+        if (!contentAndToken) {
+            return null
+        }
+        const { text, searchToken } = contentAndToken
+        const result: sourcegraph.DocumentHighlight[] = []
+        let line = 0
+        let lastLineIndex = 0
+        const tokenPattern = new RegExp('\\b' + searchToken + '\\b', 'g')
+        const matches = [...text.matchAll(tokenPattern)]
+        // TODO: why is matches empty?
```
Comment (olafurpg) [inline template/src/search/providers.ts:305]: I don't understand why `matches` is always empty here. Any advice on what I'm doing wrong? When I test the same regexp manually then it works as expected, it's only empty when running via webworkers 🤔

Comment (efritz) [reply template/src/search/providers.ts:305]: Do you know the values of text and pattern here at runtime?

Comment (olafurpg) [reply template/src/search/providers.ts:305]: Yes. I even used "copy object" on the actual logged values and `matchAll()` worked as expected against that object in the JS console. The `text` value has the full multi-line string value of the document and the pattern is correct too.

**Review by chrismwendt — APPROVED** [review]
Sweet! 🚀

**Comment by efritz** [comment]
@olafurpg Do we want to close and/or transfer the code in this PR to sg/sg?

**Comment by olafurpg** [comment]
Closing since we should implement this feature in sg/sg

**Pull request closed without merging**